### PR TITLE
add -n for path expand, so it doesn't follow symlink

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Handle for #6239, although we can change the config file to fix the issue

but refer to the [comment](https://github.com/nushell/nushell/pull/6250#issuecomment-1207263431):

> btw, this is not only an issue with nix in specific. whenever someone has a symlink that is dynamically changed, this issue will arise.

I think that might be a valid user case for `path expand`

# Tests

Make sure you've done the following:

- [X] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [X] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [X] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

File: crates/nu-command/src/path/expand.rs [hunk 1]
```diff
@@ -34,6 +35,11 @@ impl Command for SubCommand {
                 "Throw an error if the path could not be expanded",
                 Some('s'),
             )
+            .switch(
+                "no-deference",
```
Comment (fdncred) [inline crates/nu-command/src/path/expand.rs:39]: not sure what no-deference is. maybe you meant no-dereference?

Comment (WindSoilder) [reply crates/nu-command/src/path/expand.rs:39]: > maybe you meant no-dereference

Yeah, I just copy from `cp` command: https://github.com/nushell/nushell/blob/main/crates/nu-command/src/filesystem/cp.rs#L60

Comment (WindSoilder) [reply crates/nu-command/src/path/expand.rs]: I'd vote for `no-follow-link`

Comment (fdncred) [reply crates/nu-command/src/path/expand.rs]: I'm sorry but `no-follow-link` just sounds bad too. I think `dont-follow-link` sounds better, even if we leave the apostrophe out. What do you think?

**Comment by kubouch** [comment]
Thank! This is useful. I'd call the flag `--no-symlink`, `-n`, `"Do not resolve symbolic links"`. Or choose fdncred's suggestion, but make sure the short flag and the description all match.

Also, it should still follow the `--strict` flag. With your PR, it would return the path if it doesn't exist even with the `--strict` flag.

File: crates/nu-command/src/filesystem/cp.rs [hunk 2]
```diff
@@ -57,9 +57,9 @@ impl Command for Cp {
             // .switch("force", "suppress error when no file", Some('f'))
             .switch("interactive", "ask user to confirm action", Some('i'))
             .switch(
-                "no-dereference",
+                "no-symlink",
                 "If the -r option is specified, no symbolic links are followed.",
```
Comment (kubouch) [inline crates/nu-command/src/filesystem/cp.rs:61]: The -r flag here should be fixed as well.

Comment (WindSoilder) [reply crates/nu-command/src/filesystem/cp.rs]: The main purpose is that it's only useful when -r is active, I changed the description

**Comment by fdncred** [comment]
thanks

**Pull request merged**